x-pack/plugins/apm/public/components/shared/time_comparison/index.tsx

```diff
@@ -23,6 +23,8 @@ const PrependContainer = styled.div`
   padding: 0 ${px(unit)};
 `;
 
+const weekDurationInHours = moment.duration(7, 'days').asHours();
```

Comment:
nit: There's always the vanilla option. Might also be the simplest to read:
```suggestion
const weekDurationInHours = 24 * 7
```

Reply:
IMHO we should consider using moment here, but scoped to the current time, to deal with issues like DST. E.g.:

```ts
const aDayAgo = momentEnd.subtract(1, 'days').getTime();
const isAtLeastADayAgo = momentStart.getTime() <= aDayAgo;
```

```ts
const aWeekAgo = momentEnd.subtract(1, 'week').getTime();
const isAtLeastAWeekAgo = momentStart.getTime() <= aWeekAgo;
```

(we should figure out how exactly moment.subtract() deals with DST ofc)

![image](https://user-images.githubusercontent.com/352732/106649190-84f14600-6591-11eb-965b-1d3e92cb9514.png)

Reply:
I'll admit I'm not quite sure why we are using hours here, are we rounding?

Reply:
Btw, should this not be "a day before" and "a week before" (the current time range), rather than yesterday/a week ago?

Reply:
> Btw, should this not be "a day before" and "a week before" (the current time range), rather than yesterday/a week ago?

Good point. The current copy is only applicable if the end range is _now_. 
I think we could keep the labels if we add a check like:
```ts
const = end === Date.now();
if (dateDiffInHours <= 24 && isEndNow) {
```

We might have to add some padding like:
```ts
const isEndNow = end + 1000 >= Date.now();
if (dateDiffInHours <= 24 && isEndNow) {
  return [yesterdayOption, aWeekAgoOption];
}
```

If `isEndNow` is false we'll fall back to `prevPeriodOption` which might be ok.

Reply:
I don't think that will ever return true though. End is rounded. But maybe we can use the query parameter? E.g. if rangeTo=='now'

Reply:
> Btw, should this not be "a day before" and "a week before" (the current time range), rather than yesterday/a week ago?

Good catch! And I agree, the more clear we can be with the relative time the easier it is to parse for the user. So if we can add a check as @sqren proposes, I'm very happy 👍

Reply:
> End is rounded.

That's why I thought we could add padding. But since rounding is up to 5 minutes we'd need a lot of padding so I agree, not a great solution.

> But maybe we can use the query parameter? E.g. if rangeTo=='now'

Much better idea. Let's try to do that.

Reply:
you can pass a third argument, a boolean, to diff to get a floating point number instead of an integer. then you could just use <= 1. https://jsfiddle.net/8whg49qj/

Reply:
thanks will do it